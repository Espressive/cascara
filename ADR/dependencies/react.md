# React

Our React version is tied to FE's version, we can bump only after they are doing bumping Ract. For now, all the following ecosystem members will be pinned to their current version:

- @types/react v16.8.0
- @types/react-dom v16.8.0
- react v16.13.1
- react-dom v16.13.1
- react-router v5.2.0
- react-router-dom v5.2.0
- react-scripts v4.0.3
- react-test-renderer v16.13.1
