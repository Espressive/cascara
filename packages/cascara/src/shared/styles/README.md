# Cascara SASS

This will likely be only a collection of consistent styling patterns and mixins we want to use for the library. We are using SASS so we do not end up with vendor lock-in for our styles. There are dozens of options for CSS in JS out there. Some make it, and some do not. The risk is too great for us in trying to support one of these technologies for a platform like our design system.

We are using the latest version of Dart Sass. [Please review all documentation(https://sass-lang.com/documentation)] as there are some minor differences with LibSass and Ruby Sass which are older.

At some point, maybe it makes sense to move all tokens and build tooling for tokens here and make into a package to more easily include in FDS components. That time may come soon as we build more and see the limitations.
