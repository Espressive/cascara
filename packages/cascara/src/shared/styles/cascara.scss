@forward 'lib/scroll' as scroll-*;
@forward 'color/brand';

$test-var: var(--test);
