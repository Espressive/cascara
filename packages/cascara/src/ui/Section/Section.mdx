---
title: Section
propTable: Section.js
---

```jsx
<Section
  title='With Pre, Post and Sub'
  titlePost={<button type='button'>Post</button>}
  titlePre={<button type='button'>Pre</button>}
  titleSub='Some Sub Content'
>
  {<div>Content</div>}
</Section>
```
