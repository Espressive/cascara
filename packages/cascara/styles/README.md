# Cascara SASS

Collection of variables and mixins for consistent styling.

## Getting Started

### Usage

To access the variables and mixins in this package, you must declare the library in your SCSS file.

```
@use '@espressive/cascara/styles' as *;
```

> This will make all variables and mixins available in the file you are working on. The `as *` is important as it makes everything in the library available without a namespace.

### Requirements

We are using the latest version of Dart Sass. [Please review all documentation(https://sass-lang.com/documentation)] as there are some minor differences with LibSass and Ruby Sass which are older.

## What's Available

### Variables

All variables that are created from `@espressive/design-tokens` are available once declared with `@use`.

### Mixins & Helpers

There are a number of helpers available with scopes.

#### Media Queries

```
.ResponsiveModule:before {
  content: "tiny";

  @include media-xs {
    content: "small";
  }

  @include media-md {
    content: "medium";
  }

  @include media-lg {
    content: "large";
  }

  @include media-xl {
    content: "extra large";
  }
}
```

#### Scroll

```
.ScrollingModule {
  @include scroll-all;
  // or
  @include scroll-horizontal;
  // or
  @include scroll-vertical;
}
```
