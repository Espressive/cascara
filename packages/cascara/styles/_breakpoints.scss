// @use '@espressive/design-tokens/scss' as *;
// Bug in design tokens causing PX vals to get stripped and replaced with rem without conversion :/

$size-screen-0: 320px;
$size-screen-1: 768px;
$size-screen-2: 992px;
$size-screen-3: 1400px;
$size-screen-4: 1900px;

// These functions should not be needed once our library compiles correctly
@function em($number) {
  @return ($number/16)+0em;
}

@function emFromPx($number) {
  @if type-of($number) == 'number' and not unitless($number) {
    @return em($number / ($number * 0 + 1));
  }

  @return em($number);
}

// Mobile first queries (greater than)
@mixin gt-xs {
  @media screen and (min-width: #{emFromPx($size-screen-0 + 1px)}) {
    @content;
  }
}

@mixin gt-sm {
  @media screen and (min-width: #{emFromPx($size-screen-1 + 1px)}) {
    @content;
  }
}

@mixin gt-md {
  @media screen and (min-width: #{emFromPx($size-screen-2 + 1px)}) {
    @content;
  }
}

@mixin gt-lg {
  @media screen and (min-width: #{emFromPx($size-screen-3 + 1px)}) {
    @content;
  }
}

@mixin gt-xl {
  @media screen and (min-width: #{emFromPx($size-screen-4 + 1px)}) {
    @content;
  }
}

// Queries for only one breakpoint
@mixin xs {
  @media screen and (max-width: #{emFromPx($size-screen-0)}) {
    @content;
  }
}

@mixin sm {
  @media screen and (min-width: #{emFromPx($size-screen-0 + 1px)}) and (max-width: #{emFromPx($size-screen-1)}) {
    @content;
  }
}

@mixin md {
  @media screen and (min-width: #{emFromPx($size-screen-1 + 1px)}) and (max-width: #{emFromPx($size-screen-2)}) {
    @content;
  }
}

@mixin lg {
  @media screen and (min-width: #{emFromPx($size-screen-2 + 1px)} ) and (max-width: #{emFromPx($size-screen-3)}) {
    @content;
  }
}

@mixin xl {
  @media screen and (min-width: #{emFromPx($size-screen-3 + 1px)}) and (max-width: #{emFromPx($size-screen-4)}) {
    @content;
  }
}
