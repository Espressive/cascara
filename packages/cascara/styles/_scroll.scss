@use '~@espressive/design-tokens/scss' as *;

%scrollbars {
  -webkit-overflow-scrolling: touch;
  scroll-behavior: smooth;

  &::-webkit-scrollbar {
    width: 0;
  }

  &::-webkit-scrollbar-thumb {
    background-clip: content-box;
    background-color: $color-base-gray-medium;
    border: 0.25rem solid rgba(0, 0, 0, 0);
    border-radius: 1rem;
    box-shadow: inset 0 0 0 0.05em rgba(255, 255, 255, 0.2);
  }

  &:hover {
    &::-webkit-scrollbar {
      width: 0.875rem;
    }
  }
}

@mixin vertical {
  overflow-y: overlay;

  @extend %scrollbars;
}

@mixin horizontal {
  overflow-x: overlay;

  @extend %scrollbars;
}

@mixin all {
  overflow: overlay;

  @extend %scrollbars;
}
