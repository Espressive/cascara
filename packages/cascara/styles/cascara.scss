@forward '@espressive/design-tokens/scss';
@forward 'breakpoints' as media-*;
@forward 'scroll' as scroll-*;
