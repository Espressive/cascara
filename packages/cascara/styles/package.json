{
  "name": "@espressive/cascara/styles",
  "private": true,
  "sideEffects": false,
  "main": "cascara.scss",
  "css": "cascara.scss"
}
