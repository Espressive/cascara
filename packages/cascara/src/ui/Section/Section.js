import React from 'react';
import pt from 'prop-types';

import styles from './Section.module.scss';

const propTypes = {
  children: pt.oneOfType([pt.node, pt.arrayOf(pt.node)]),
  footer: pt.oneOfType([pt.element, pt.arrayOf(pt.element)]),
  header: pt.oneOfType([pt.element, pt.arrayOf(pt.element)]),
};

const Section = ({ children, footer, header }) => {
  return (
    <section className={styles._}>
      {header && <header className={styles.Header}>{header}</header>}
      <div className={styles.Content}>{children}</div>
      {footer && <footer className={styles.Footer}>{footer}</footer>}
    </section>
  );
};

Section.propTypes = propTypes;

export default Section;
