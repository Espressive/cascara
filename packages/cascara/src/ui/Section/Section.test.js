import { render } from '@testing-library/react';

import fixtures from './Section.fixture';

describe('section', () => {
  //Run snapshot test for all fixtures
  test('snapshot tests', async () => {
    for (const fixture in fixtures) {
      const view = render(fixtures[fixture]).container;

      expect(view).toMatchSnapshot();
    }
  });
});
