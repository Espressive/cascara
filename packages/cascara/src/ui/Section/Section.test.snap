// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section snapshot tests 1`] = `
<div>
  <section
    class="Section isBasic"
  >
    <div
      class="_ _ fluid space-start vAlign-start"
    >
      <button
        type="button"
      >
        Pre
      </button>
      <h3
        class="Title"
      >
        With Pre, Post and Sub
         
        <div
          class="Sub"
        >
          Some Sub Content
        </div>
      </h3>
      <div
        class="Item push"
      >
        <button
          type="button"
        >
          Post
        </button>
      </div>
    </div>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 2`] = `
<div>
  <section
    class="Section"
  >
    <div
      class="_ _ fluid space-start vAlign-start"
    >
      <button
        type="button"
      >
        Pre
      </button>
      <h3
        class="Title"
      >
        With Pre, Post and Sub
         
        <div
          class="Sub"
        >
          Some Sub Content
        </div>
      </h3>
      <div
        class="Item push"
      >
        <button
          type="button"
        >
          Post
        </button>
      </div>
    </div>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 3`] = `
<div>
  <section
    class="Section"
  />
</div>
`;

exports[`section snapshot tests 4`] = `
<div>
  <section
    class="Section"
  >
    <h3
      class="Title"
    >
      Only Section Markup
       
    </h3>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 5`] = `
<div>
  <section
    class="Section"
  >
    <div
      class="_ _ fluid space-start vAlign-start"
    >
      <button
        type="button"
      >
        Pre
      </button>
      <h3
        class="Title"
      >
        With Pre
         
      </h3>
    </div>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 6`] = `
<div>
  <section
    class="Section"
  >
    <div
      class="_ _ fluid space-start vAlign-start"
    >
      <h3
        class="Title"
      >
        With Post
         
      </h3>
      <div
        class="Item push"
      >
        <button
          type="button"
        >
          Post
        </button>
      </div>
    </div>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 7`] = `
<div>
  <section
    class="Section"
  >
    <div
      class="_ _ fluid space-start vAlign-start"
    >
      <button
        type="button"
      >
        Pre
      </button>
      <h3
        class="Title"
      >
        With Pre and Post
         
      </h3>
      <div
        class="Item push"
      >
        <button
          type="button"
        >
          Post
        </button>
      </div>
    </div>
    <div>
      Content
    </div>
  </section>
</div>
`;

exports[`section snapshot tests 8`] = `
<div>
  <section
    class="Section"
  >
    <h3
      class="Title"
    >
      With Sub Section
       
      <div
        class="Sub"
      >
        Some Sub Content
      </div>
    </h3>
    <div>
      Content
    </div>
  </section>
</div>
`;
